apollo-compiler/src/main/kotlin/com/apollographql/apollo/compiler/backend/ir/BackendIr.kt

```diff
@@ -0,0 +1,116 @@
+package com.apollographql.apollo.compiler.backend.ir
+
+import com.apollographql.apollo.compiler.parser.introspection.IntrospectionSchema
+
+internal data class SelectionKey(
+    val root: String,
+    val keys: List<String>,
+    val type: Type,
+) {
+  operator fun plus(name: String): SelectionKey {
+    return this.copy(
+        keys = this.keys + name
+    )
+  }
+
+  enum class Type {
+    Query, Fragment
+  }
+}
+
+internal data class BackendIr(
+    val operations: List<Operation>,
+    val fragments: List<NamedFragment>,
+    val typeDeclarations: List<IntrospectionSchema.TypeRef>,
+    val typesPackageName: String,
+    val fragmentsPackageName: String,
+) {
+  data class Operation(
+      val name: String,
+      val operationName: String,
+      val targetPackageName: String,
+      val operationType: IntrospectionSchema.TypeRef,
+      val comment: String,
+      val variables: List<Variable>,
+      val definition: String,
+      val dataField: Field,
+  )
+
+  data class Variable(
+      val name: String,
+      val type: IntrospectionSchema.TypeRef,
+  )
+
+  data class Field(
+      val name: String,
+      val alias: String?,
+      val type: IntrospectionSchema.TypeRef,
+      val args: List<Argument>,
+      val fields: List<Field>,
+      val fragments: List<InlineFragment>,
+      val conditions: List<Condition>,
+      val description: String,
+      val deprecationReason: String?,
+      val selectionKeys: Set<SelectionKey>,
```

Comment:
Shouldn't there be a list of named fragments there as well? Or do you map the named fragments to inline fragments since they're quite similar?